include/openthread/tcp.h

```diff
@@ -0,0 +1,616 @@
+ * @param[in]  aReceiveAvailableCallback  The new callback.
+ *
+ * @retval OT_ERROR_NONE    Successfully set the callback.
+ * @retval OT_ERROR_FAILED  Failed to set the callback.
+ */
+otError otTcpSetReceiveAvailableCallback(otInstance *aInstance, otTcpEndpoint *aEndpoint, otTcpReceiveAvailable aReceiveAvailableCallback);
+
+/**
+ * Sets the "disconnected" callback on a TCP endpoint.
+ *
+ * @param[in]  aInstance                A pointer to the OpenThread instance to which the TCP endpoint belongs.
+ * @param[in]  aEndpoint                A pointer to the TCP endpoint structure whose callback to set.
+ * @param[in]  aDisconnectedCallback    The new callback.
+ *
+ * @retval OT_ERROR_NONE    Successfully set the callback.
+ * @retval OT_ERROR_FAILED  Failed to set the callback.
+ */
+otError otTcpSetDisconnectedCallback(otInstance *aInstance, otTcpEndpoint *aEndpoint, otTcpDisconnected aDisconnectedCallback);
+
+/**
+ * Binds the TCP endpoint to an IP address and port.
+ *
+ * @param[in]  aInstance   A pointer to the OpenThread instance to which the TCP endpoint belongs.
+ * @param[in]  aEndpoint   A pointer to the TCP endpoint structure to bind.
+ * @param[in]  aSockName   The address and port to which to bind this TCP endpoint.
+ *
+ * @retval OT_ERROR_NONE    Successfully bound the TCP endpoint.
+ * @retval OT_ERROR_FAILED  Failed to bind the TCP endpoint.
+ */
+otError otTcpBind(otInstance *aInstance, otTcpEndpoint *aEndpoint, const otSockAddr *aSockName);
+
+/**
+ * This enumeration defines flags passed to @p otTcpConnect.
+ */
+enum {
+    OT_TCP_CONNECT_NO_FAST_OPEN = 1 << 0,
+};
+
+/**
+ * Records the remote host and port for this connection. By default TCP Fast
+ * Open is used. This means that this function merely records the remote host
+ * and port, and that the TCP connection establishment handshake only happens
+ * on the first call to @p otTcpSendByReference. TCP Fast Open can be
+ * explicitly disabled using @p aFlags, in which case the TCP connection
+ * establishment handshake is initiated immediately.
+ *
+ * @param[in]  aInstance  A pointer to the OpenThread instance to which the TCP endpoint belongs.
+ * @param[in]  aEndpoint  A pointer to the TCP endpoint structure to connect.
+ * @param[in]  aSockName  The IP address and port of the host to which to connect.
+ * @param[in]  aFlags     Flags specifying options for this operation (see enumeration above).
+ *
+ * @retval OT_ERROR_NONE    Successfully completed the operation.
+ * @retval OT_ERROR_FAILED  Failed to complete the operation.
+ */
+otError otTcpConnect(otInstance *aInstance, otTcpEndpoint *aEndpoint, const otSockAddr *aSockName, uint32_t aFlags);
+
+/**
+ * This enumeration defines flags passed to @p otTcpSendByReference.
+ */
+enum {
+    OT_TCP_SEND_MORE_TO_COME = 1 << 0,
+};
+
+/**
+ * Adds data referenced by the linked buffer pointed to by @p aBuffer to the
```

Comment:
nit: have one-line brief in own block

Reply:
Done. I'm not 100% what you meant by "in own block", can you please check that what I've done is acceptable?